op-service/txmgr/queue.go

```diff
@@ -51,17 +51,36 @@ func (q *Queue[T]) Wait() error {
 	return q.group.Wait()
 }
 
+// HandleResponse will wait for the response on the first passed channel,
+// and then forward it on the second passed channel (attaching the id). It returns
+// the response error or the context error if the context is canceled.
+func HandleResponse[T any](ctx context.Context, c chan SendResponse, d chan TxReceipt[T], id T) error {
+	select {
+	case response := <-c:
```

Comment:
Is there any concern that this channel could be closed, causing the response to be blank?

Reply:
The existing code that handles sending on the channel (which is in the same package) never closes it, so I don't think it is a concern right now. 

To defend against future changes, we could of course do 

```go
case response, ok := <-c
```
and then handle the `!ok` case. WDYT? 

We could panic if the channel were ever closed, this would count as a programming error IMO.

Reply:
On reflection I don't think this is necessary to do.